introduction_to_applying_machine_learning/deepar_chicago_traffic_violations/deepar_chicago_traffic_violations.ipynb

```diff
@@ -0,0 +1,467 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "\n",
+    "sess = sagemaker.Session()\n",
+    "bucket = sess.default_bucket()\n",
+    "base = 'DEMO-deepar-chicago' # notebook auther to input a short descriptive name\n",
+    "prefix = 'sagemaker/' + base\n",
+    "region = sess.boto_region_name\n",
+    "\n",
+    "role = sagemaker.get_execution_role()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we'll import the Python library like s3fs, matplotlib, pandas and numpy"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "import requests, io\n",
+    "import boto3\n",
+    "import numpy as np\n",
+    "!conda install -y s3fs\n",
+    "np.random.seed(1)\n",
+    "import pandas as pd\n",
+    "import json\n",
+    "import matplotlib.pyplot as plt\n",
+    "import s3fs\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "## Data\n",
+    "\n",
+    "\n",
+    "\n",
+    "The speed violations are captured by camera systems and available to improve the lives of public from the city of Chicago data portal. The Speed Camera Violation data set can be used to discern patterns in the data and gain meaningful insights.\n",
+    "\n",
+    "The data set contains multiple camera locations and daily violation counts. If we imagine that each daily violations for a camera as one time series, we can use Amazon SageMaker’s DeepAR algorithm to train a model for multiple streets simultaneously, and predict violation for multiple street cameras using the Amazon SageMaker’s DeepAR algorithm.\n",
+    "\n",
+    "The dataset contains several columns, we will use the Address, violation date, violations for the forecasting.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datafile = 'Chicago_Speed_Camera_Violations.csv'\n",
+    "speeding_violation_data_path = '{}/{}/{}.csv'.format(bucket,prefix, datafile)\n",
+    "s3_output_path = '{}/{}/output'.format(bucket, prefix)\n",
+    "s3_data_path = '{}/{}/data'.format(bucket, prefix)\n",
+    "train_data_path = '{}/{}/train/train.json'.format(bucket, prefix)\n",
+    "test_data_path = '{}/{}/test/test.json'.format(bucket, prefix)\n",
+    "\n",
+    "url = 'https://data.cityofchicago.org/api/views/hhkd-xvj4/rows.csv?accessType=DOWNLOAD'\n",
+    "\n",
+    "# get the data from City of Chicago site\n",
+    "r = requests.get(url, allow_redirects=True)\n",
+    "open(datafile, 'wb').write(r.content)\n",
+    "\n",
+    "# store it in S3 object\n",
+    "key = '{}/{}'.format(prefix, datafile)\n",
+    "s3client = boto3.client('s3', region_name=region)\n",
```

Comment:
[nitpick] it would be more Pythonic  to name the variable `s3_client`

Reply:
fixed